src/devices/GoPiGo3/Movements/Motor.cs

```diff
@@ -0,0 +1,199 @@
+            _goPiGo.SetMotorPower(Port, speed);
+        }
+
+        /// <summary>
+        /// Set Tachometer encoder offset
+        /// Use this to reset or setup a specific position
+        /// </summary>
+        /// <param name="position">New offset, 0 to reset</param>
+        public void SetTachoCount(int position)
+        {
+            _goPiGo.SetMotorPosition(Port, position);
+        }
+
+        /// <summary>
+        /// Stop the Motor
+        /// </summary>
+        public void Stop()
+        {
+            _goPiGo.SetMotorPower(Port, 0);
+        }
+
+        /// <summary>
+        /// Start the motor
+        /// </summary>
+        public void Start()
+        {
+            _goPiGo.SetMotorPower(Port, Speed);
+        }
+
+        /// <summary>
+        /// Start with the specified speed
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        public void Start(int speed)
+        {
+            SetSpeed(speed);
+            Start();
+        }
+
+        /// <summary>
+        /// Change the polatity of the motor
+        /// </summary>
+        /// <param name="polarity">Polarity of the motor, backward, forward or opposite</param>
+        public void SetPolarity(Polarity polarity)
+        {
+            try
+            {
+                var motorstatus = _goPiGo.GetMotorStatus(Port);
+                switch (polarity)
+                {
+                    case Polarity.Backward:
+                        if (motorstatus.Speed > 0)
+                            _goPiGo.SetMotorPower(Port, -Speed);
+                        break;
+                    case Polarity.Forward:
+                        if (motorstatus.Speed < 0)
+                            _goPiGo.SetMotorPower(Port, -Speed);
+                        break;
+                    case Polarity.OppositeDirection:
+                        _goPiGo.SetMotorPower(Port, -Speed);
+                        break;
+                    default:
+                        break;
+                }
+            }
+            catch (Exception) { }
+        }
+
+        /// <summary>
+        /// Gets the tacho count
+        /// </summary>
+        /// <returns>The tacho count. To get the tick per degree, check GoPiGo3.MotorTicksPerDegree property</returns>
+        public int GetTachoCount()
+        {
+            try
+            {
+                return _goPiGo.GetMotorEncoder(Port);
+            }
+            catch (Exception)
+            {
+                return int.MaxValue;
+            }
+        }
+
+        /// <summary>
+        /// Get the speed
+        /// </summary>
+        /// <returns>speed is between -255 and +255</returns>
+        public int GetSpeed()
+        {
+            return _goPiGo.GetMotorStatus(Port).Speed;
+        }
+
+        /// <summary>
+        /// Set or read the speed of the motor
+        /// speed is between -255 and +255
+        /// </summary>
+        public int Speed
```

Comment:
I'd make Get/Set functions private, also please split this into separate lines

Reply:
I prefer to leave the Speed, the Get and Set speed functions all public. It gives choice to use either the property either the functions.